### PR TITLE
ESQL: Move data types into core

### PR DESCRIPTION
**Pull request by nik9000**
This moves all of the new data types declared in `EsqlDataTypes` into `DataTypes`. It also removes `EsqlDataTypes#types` and makes `DataTypes#types` return what that used to return. It doesn't modify any other methods of `EsqlDataTypes`. That's a change for another time.

### CODE REVIEW

**Review by alex-spies — APPROVED** [review]
Thanks, @nik9000 !

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataTypes.java [hunk 1]
```diff
@@ -48,8 +48,6 @@ public final class DataTypes {
     public static final DataType IP               = new DataType("ip",                45,                false, false, true);
     // version
     public static final DataType VERSION          = new DataType("version",           Integer.MAX_VALUE, false, false, true);
-    // binary
-    public static final DataType BINARY           = new DataType("binary",            Integer.MAX_VALUE, false, false, true);
```
Comment (not-napoleon) [inline x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataTypes.java:52]: Why are we removing Binary? is it just never used in ESQL?

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataTypes.java
Comment (not-napoleon) [inline x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataTypes.java:1]: Should we move this out of `esql-core` and into main line esql, while we're at it?

**Review by luigidellaquila — APPROVED** [review]
LGTM, thanks!

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataTypes.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataTypes.java:52]: Yeah, never used at all.

File: x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataTypes.java
Comment (nik9000) [reply x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataTypes.java:1]: We could. If we do that we're going to have to be careful about serialization. We can though.

**Pull request merged**